src/diggler/command.d

```diff
@@ -422,22 +422,66 @@ abstract class CommandSet(T) : ICommandSet
 
 	final void registerCommands(T cmdSet)
 	{
-		foreach(memberName; __traits(derivedMembers, T))
+		import std.meta : Filter;
+		import std.algorithm;
+		import std.string;
+		import std.array;
+
+		enum isCommand(string symbol) =
+			symbol != "__ctor" &&
+			symbol != "__dtor" &&
+			__traits(compiles, __traits(getMember, T, symbol)) && // ahem...
+			__traits(getProtection, __traits(getMember, T, symbol)) == "public" &&
+			isSomeFunction!(mixin("T." ~ symbol)) &&
+			!__traits(isStaticFunction, mixin("T." ~ symbol)) &&
+			!hasAttribute!(mixin("T." ~ symbol), ignore);
+
+		enum isSub(string symbol) = symbol[0] != '_' && symbol.canFind("_");
```

Comment:
The underscore should be customizable by the command set, in case the user wishes to use `_` in command names. e.g. `T.subCommandSeparator`.